generator/python/resources/templates/rest.mustache

```diff
@@ -292,8 +292,8 @@ class RESTClientObject(object):
 
         if not missing_credentials and missing_or_expired_token:
             self.token = self.call_auth_endpoint(headers)
-            return True
-        return False
+
+        return self.token is not None
```

Comment:
This may return true when missing_or_expired_token is true and missing_credentials is true, in which case we'd better return False as it was before.


So maybe something we could get rid of the missing_or_expired variable and just do something like that:

```
  if not self.token.is_valid_enough():
    self.token = None

  if not missing_credentials and self.token = None:
    self.token = self.call_auth_endpoints(headers)

  return self.token is not None
```

Reply:
Kind of done, I decided to switch the refresh_token() method output so that it directly returns an access_token as a string. So that the logic the refresh logic is isolated to the method and not spread between the method and the Token object.